homeassistant/components/sensor/__init__.py

```diff
@@ -221,10 +222,6 @@ def unit_of_measurement(self) -> str | None:
     @property
     def state(self) -> Any:
         """Return the state of the sensor and perform unit conversions, if needed."""
-        # Test if _attr_state has been set in this instance
-        if "_attr_state" in self.__dict__:
-            return self._attr_state
-
```

Comment:
This will break custom components using `_attr` shorthand, should we instead log a warning if we hit this code?

Reply:
Do we need to rush to remove it?

Reply:
Both the changes in this PR were suggested in review comments in #48261, I just want to make sure that's taken care of.
I don't think there's any rush.

Reply:
It has only been in there since late May, we have the blogpost and if we also mention this in the release notes I think this should be fine